MIR-1343 Do not render mir-admindata-box staticly

Possommi opened the pull request:
[MIR-1343](https://mycore.atlassian.net/browse/MIR-1343) and [MIR-1341](https://mycore.atlassian.net/browse/MIR-1341)

requires https://github.com/MyCoRe-Org/mycore/pull/2215 to render the dates properly


[MIR-1343]: https://mycore.atlassian.net/browse/MIR-1343?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MIR-1341]: https://mycore.atlassian.net/browse/MIR-1341?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Possommi commented:
@toKrause should https://mycore.atlassian.net/browse/MIR-1341 included in this PR

toKrause commented:
> @toKrause should [MIR-1341](https://mycore.atlassian.net/browse/MIR-1341) included in this PR?

Would be okay for me and make things easier, organizationally.

toKrause commented:
Code looks okay and I've tested the modified feature in my development system. 

One thing needs to be considered: Once the changes in this PR are applied, already existing statically cached content (that still contains HTML) is unsuitable to be rendered with the updated style sheets. It will look something like this:

![Screenshot 2024-08-01 at 12-52-28 Habilitation](https://github.com/user-attachments/assets/527625e4-cb25-4767-a0c5-15db17f56a64)

After any change, the newly created statically cached content (that now contains XML) works fine. It will look something like this: 

![image](https://github.com/user-attachments/assets/32c5409f-06c6-41bb-b7c1-fac07fd051ce)

Since this PR targets the LTS branch, some migration strategy has to be devised and implemented. Ideally, one, that is going to be performed automatically. 

What are your plans here?

Once a migration strategy has been devised and implemented, I will approve this PR.

Possommi commented:
> Code looks okay and I've tested the modified feature in my development system.
> [...]
> Since this PR targets the LTS branch, some migration strategy has to be devised and implemented. Ideally, one, that is going to be performed automatically.
> 
> What are your plans here?
> 
> Once a migration strategy has been devised and implemented, I will approve this PR.

[1] Executing `repair metadata search of type mods` would update all static content. 

[2] Alternatively I could add a check in `mods-metadata-page.xsl`
```
<div id="modalFrame-body" class="modal-body">
  <xsl:choose>
    <xsl:when test="$originalContent/div[@id=$boxID]/table">
      <xsl:copy-of select="$originalContent/div[@id=$boxID]/*"/>
       ...
      <xsl:otherwise>     
        <xsl:apply-templates select="$originalContent/div[@id=$boxID]/*" mode="history-modal"/>
      ...
</div>
``` 
I do prefer [1] although [2] requires no additional user actions after upgrading to the new LTS.

toKrause commented:
Version [1] would certainly be simpler and would be okay (if documented in the migration documentation), if this PR would target the main branch, but since this PR targets the LTS branch, there really should be no manual migration step.

So, version [2] should probably be implemented.

toKrause reviewed (changes requested):
Version [1] would certainly be simpler and would be okay (if documented in the migration documentation), if this PR would target the main branch, but since this PR targets the LTS branch, there really should be no manual migration step.

So, version [2], if I understand it correctly, wold render an old static file in the old way and a new  static file in the new way. I don't think that is a desirable solution, since it would be unclear to the user why the history has (seemingly an a random basis) has a different behavior for different objects. I think a solution should be invented, that automatically creates the missing/unfit static data for in the new way if no/old static data is found.

toKrause commented on mir-module/src/main/java/org/mycore/mir/migration/MIRMigrateStaticHistoryContentJobAction.java:
```diff
@@ -0,0 +1,89 @@
+        logger = LogManager.getLogger(MIRMigrateStaticHistoryContentJobAction.class);
+
+        staticHistoryPath = Path.of(
+            MCRConfiguration2.getString("MCR.Object.Static.Content.Default.Path").get() + File.separator
```
Should use `MCRConfiguration2#getStringOrThrow`

toKrause commented on mir-module/src/main/java/org/mycore/mir/migration/MIRMigrateStaticHistoryContent.java:
```diff
@@ -0,0 +1,66 @@
+
+    @Override
+    public int getPriority() {
+        return 0;
```
Should probably have a lower priority, since `org.mycore.services.queuedjob.MCRJobQueueInitializer` also uses priority `0` and this `AutoExecutable` relies on the job queue to be initialized.
